k2/csrc/tensor.cu

```diff
@@ -131,9 +132,57 @@ void Tensor::Init(ContextPtr c) {
 }
 
 Tensor ToContiguous(const Tensor &tensor) {
-  // TODO(haowen): implement
-  K2_LOG(FATAL) << "Not implemented";
-  return tensor;
+  if (tensor.IsContiguous()) return tensor;
+  std::vector<int32_t> dims = tensor.Dims();
+  std::vector<int32_t> strides = tensor.Strides();
+  int32_t num_dims = static_cast<int32_t>(dims.size());
+
+  std::size_t num_dims_bytes = num_dims * sizeof(int32_t);
+
+  RegionPtr dims_ptr = NewRegion(tensor.Context(), num_dims_bytes);
+  RegionPtr strides_ptr = NewRegion(tensor.Context(), num_dims_bytes);
+
+  auto cpu_context = tensor.Context()->GetCpuContext();
```

Comment:
This is a bit heavy-handed for just transferring dims and strides.  The Shape object has a fixed size with no pointers.  I suggest you give it a device constructor and pass it in as a captured variable.
I'm not sure if it's necessary to make a local copy of it... dunno if the implicit-capture-of-this only happens
for *this* class object or for all class objects.

Reply:
And if you have to copy `std::vector` to device, you can just call
```
Array1<int32_t> dims_array(ctx, dims);
const int32_t *dims_data = dims_array.Data();
```

Reply:
Fixed (removed).